contrib/linux/actions/dig.py

```diff
@@ -39,11 +39,17 @@ def run(self, rand, count, nameserver, hostname, queryopts):
             cmd_args.append('+' + v)
 
         cmd_args.append(hostname)
-        result_list = filter(None, subprocess.Popen(cmd_args,
-                                                    stderr=subprocess.PIPE,
-                                                    stdout=subprocess.PIPE)
-                                             .communicate()[0]
-                                             .split('\n'))
+
+        try:
+            result_list = filter(None, subprocess.Popen(cmd_args,
+                                                        stderr=subprocess.PIPE,
+                                                        stdout=subprocess.PIPE)
+                                 .communicate()[0]
+                                 .split('\n'))
+        except OSError as e:
```

Comment:
I'm not a huge fan of treating all `OSError`s as `ENOENT` errors.

Python 3 has an explicit [`FileNotFoundError`](https://docs.python.org/3/library/exceptions.html#FileNotFoundError), but for now we also need to support Python 2.

There is a way to check the exception for the type of error:

```python
import errno

# ...

except OSError as e:
    if e.errno == errno.ENOENT:
        # ...
    else:
        raise e
```

Can you tweak this to only emit the error message on an `ENOENT` `OSError`?

Also, please add a `NOTE:` comment so we can revisit this once we only support Python 3.

Reply:
@blag We'll isn't that a little bit of python lore. Yes I will update this.

Reply:
@blag pushed the updates you requested.